tarodb/sqlerrors.go

```diff
@@ -0,0 +1,64 @@
+package tarodb
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgerrcode"
+	"modernc.org/sqlite"
+	sqlite3 "modernc.org/sqlite/lib"
+)
+
+// ParseSqlError attempts to interpret a given error as a database agnostic SQL
+// error.
+func ParseSqlError(err error) error {
+	// Attempt to interpret the error as a sqlite error.
+	var sqliteErr *sqlite.Error
+	if errors.As(err, &sqliteErr) {
+		return parseSqliteError(sqliteErr)
+	}
+
+	// Attempt to interpret the error as a postgres error.
+	var pqErr *pgconn.PgError
+	if errors.As(err, &pqErr) {
+		return parsePostgresError(pqErr)
+	}
+
+	// Indicate that the error could not be parsed.
+	return nil
```

Comment:
This should just return the original error! Otherwise we end up ignoring errors that we don't know. If we just return the original that also works well with using this in `ExecTx` directly. So whenever we look at an error returned by `ExecTx` we know it's already been re-mapped. And if it's not a type we re-map then it's just the original error.

Reply:
:+1: yes, when moved into `ExecTx` this is certainly necessary.